File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
        string al="";
        for(char ch:s)
        {
            if(isalnum(ch))
            {
                al+=tolower(ch);
            }
        }
        string rev=al;
        reverse(al.begin(),al.end());
        if(rev==al)
        {
            return true;
        }
        else
        {
            return false;
        }
        
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        char arr[]=s.toCharArray();
        String c="";
        for(char ch:arr)
        {
            if(Character.isLetterOrDigit(ch)){
                c+=Character.toLowerCase(ch);
            }
        }
        int i=0,j=c.length()-1;
        while(i<j)
        {
            if(c.charAt(i)!=c.charAt(j))
            {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}